lib/controllers/run-controller.ts

```diff
@@ -355,53 +341,52 @@ export class RunController extends EventEmitter implements IRunController {
 					force: liveSyncInfo.force,
 					connectTimeout: 1000
 				};
-				let liveSyncResultInfo = await platformLiveSyncService.liveSyncWatchAction(device, watchInfo);
+				const deviceAppData = await platformLiveSyncService.getAppData(_.merge({ device, watch: true }, watchInfo));
 
-				await this.refreshApplication(projectData, liveSyncResultInfo, data, deviceDescriptor);
+				if (data.hasNativeChanges) {
+					const rebuiltInfo = this.rebuiltInformation[platformData.platformNameLowerCase] && (this.$mobileHelper.isAndroidPlatform(platformData.platformNameLowerCase) || this.rebuiltInformation[platformData.platformNameLowerCase].isEmulator === device.isEmulator);
+					if (!rebuiltInfo) {
+						await this.$prepareNativePlatformService.prepareNativePlatform(platformData, projectData, prepareData);
+						await deviceDescriptor.buildAction();
+						this.rebuiltInformation[platformData.platformNameLowerCase] = { isEmulator: device.isEmulator, platform: platformData.platformNameLowerCase, packageFilePath: null };
+					}
 
-				this.emitCore(RunOnDeviceEvents.runOnDeviceExecuted, {
-					projectDir: projectData.projectDir,
-					deviceIdentifier: device.deviceInfo.identifier,
-					applicationIdentifier: projectData.projectIdentifiers[device.deviceInfo.platform.toLowerCase()],
-					syncedFiles: liveSyncResultInfo.modifiedFilesData.map(m => m.getLocalPath()),
-					isFullSync: liveSyncResultInfo.isFullSync
-				});
+					await this.$deviceInstallAppService.installOnDevice(device, deviceDescriptor.buildData, this.rebuiltInformation[platformData.platformNameLowerCase].packageFilePath);
```

Comment:
We need to add fullSync here, after install. This logic is specific for LiveSync on iOS device.